api/share/requests/users.py

```diff
@@ -0,0 +1,65 @@
+import re
+from api.share.requests import ValidRequestObject, InvalidRequestObject
+
+
+class GetUsersCountRequest(ValidRequestObject):
+
+    def __init__(self,
+                 registration_bot_chat_id: str,
+                 tg_username: str,
+                 phone: str,
+                 email: str,
+                 first_name: str,
+                 last_name: str):
+        self.registration_bot_chat_id = registration_bot_chat_id
+        self.tg_username = tg_username
+        self.phone = phone
+        self.email = email
+        self.first_name = first_name
+        self.last_name = last_name
+
+    @classmethod
+    def from_dict(cls, input_dict):
+        # Check input parameters.
+        invalid_request = InvalidRequestObject()
+        # https://core.telegram.org/bots/api#chat : chat_id - max 52 significant bits
+        # 9007199254740991 is the max number you can store in the 53 bits => len(chat_id) <= 16
+        if input_dict.get("registration_bot_chat_id") and \
+                len(input_dict.get("registration_bot_chat_id")) > 16:
+            invalid_request.add_error("registration_bot_chat_id", "Invalid length")
+
+        if input_dict.get("tg_username") and (len(input_dict.get("tg_username")) < 5 or
+                                              len(input_dict.get("tg_username")) > 32):
+            invalid_request.add_error("tg_username", "Invalid length")
+
+        # See modal PersonalInfo
+        if input_dict.get("first_name") and len(input_dict.get("first_name")) > 100:
+            invalid_request.add_error("first_name", "Invalid length")
+
+        # See modal PersonalInfo
+        if input_dict.get("last_name") and len(input_dict.get("last_name")) > 100:
```

Comment:
BTW maybe strings should be extracted into constants as well, but I'm not sure. Numbers definitely, though.

Reply:
I think such questions are good for refactoring stage

Reply:
Sure, but just keep in mind that `master` should receive code that is already refactored in this way :)

Reply:
Ok, I got it